artiq/gui/entries.py

```diff
@@ -99,6 +99,9 @@ def state_to_value(state):
     def default_state(procdesc):
         if "default" in procdesc:
             return procdesc["default"]
+        elif "min" in procdesc and procdesc["min"]:
+            # when the min value is negative, we clamp it to 0 instead
```

Comment:
Why?

Reply:
Seems not just superfluous but also incorrect - the bug will still be present if both min and max are negative.

Reply:
hmm, we can just add an additional check whether 0 is within the range of min & max, but let say we just set the minimum value here then.

Reply:
What's special about 0? Why not e.g. return ``(min+max)/2`` if we have both, or just ``min`` or ``max`` if we have only one of them?

Reply:
And please do your research about ``/`` vs. ``//`` in Python and use them appropriately.

Reply:
using midpoint as the default value...feels bit odd though, should we do it?

Reply:
Can also keep 0 iff min <= 0 < max, then it's also backward compatible.

Reply:
Or just min instead of midpoint?